rxandroidble/src/main/java/com/polidea/rxandroidble/internal/RxBleDeviceImpl.java

```diff
@@ -53,12 +54,20 @@
 
     @Override
     public Observable<RxBleConnection> establishConnection(final boolean autoConnect) {
+        ConnectionSetup options = (new ConnectionSetup.Builder())
+                .setAutoConnect(autoConnect)
+                .build();
```

Comment:
This should also have `.setSuppressOperationCheck(false)` to match the previous behaviour.